hilo.rb

```diff
@@ -1,34 +1,111 @@
 #!/usr/bin/env ruby
-puts "Welcome to the Hi-Lo game.\nGuess a random whole number from 1 to 100 (inclusive)."
-puts "You will have a total of 6 guesses."
-puts ""
 
-random_num = rand(1..100)
+# Game to guess random number
+class HiLoGame
 
-# gives the user 6 guesses
-for i in 0..5
-    guesses_left = 6 - i
-    puts "You have #{guesses_left} guesses left."
-    puts "Please guess a whole number."
+    # Constructor to initialise the class with a maximum number of guesses
+    def initialize(max_guesses)
+	@max_guesses = max_guesses
+	@already_guessed = Array.new
+	@random_num
+	@has_won = false
```

Comment:
It's not very rubyish to store a boolean to store a state that you can easily infer from other information. Like you have the guesses and you know the random number so you can just figure out if the game was won. hint: `has_won?`